public/app/core/components/ThreePaneSplit/ThreePaneSplit.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { createRef, MutableRefObject, PureComponent } from 'react';
+import SplitPane from 'react-split-pane';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { stylesFactory } from '@grafana/ui';
+import { config } from 'app/core/config';
+
+enum Pane {
+  Right,
+  Top,
+}
+
+interface Props {
+  horizontalSplitComponents: React.ReactNode[];
+  rightPaneComponent: React.ReactNode;
+  uiState: { topPaneSize: number | string; rightPaneSize: number | string };
```

Comment:
looks like we always treat it as a number, when is it string?

Reply:
It takes the topPaneSize and rightPaneSize from this model  https://github.com/grafana/grafana/blob/7d8cb6869a2d3cb2bf3dd54dac2b59e60babc975/public/app/features/dashboard/components/PanelEditor/state/reducers.ts#L20

Reply:
but then we don't handle that case in the component :thinking: